sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -125,10 +163,22 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
     val stagingDir = hadoopConf.get("hive.exec.stagingdir", ".hive-staging")
     val scratchDir = hadoopConf.get("hive.exec.scratchdir", "/tmp/hive")
 
+    // Hive sets session_path as HDFS_SESSION_PATH_KEY(_hive.hdfs.session.path) in hive config
+    val sessionScratchDir = externalCatalog.unwrapped.asInstanceOf[HiveExternalCatalog]
+      .client.getConf("_hive.hdfs.session.path", "")
```

Comment:
Shall we have an assert on it or throw a proper exception when `_hive.hdfs.session.path` was not set?

Reply:
If `_hive.hdfs.session.path` is empty, `getMRTmpPath` uses `scratchDir` instead of `sessionScratchDir` in this line:  `    val sessionPath = Option(sessionScratchDir).filterNot(_.isEmpty).getOrElse(scratchDir)`.

Reply:
It is tested in this unit test.
https://github.com/apache/spark/pull/27690/files#diff-ee422d26750ba346c81b7f85b4b14577R46

Reply:
Thanks, @moomindani. How does Hive behaves when `_hive.hdfs.session.path` is not set?

Reply:
As you mentioned in another comment, empty string is retrieved when `_hive.hdfs.session.path` is not set.